src/Core/Domain/Tax/ValueObject/TaxStatus.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Tax\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Tax\Exception\TaxConstraintException;
+
+/**
+ * Provides tax status data
+ */
+class TaxStatus
```

Comment:
@matks wdyt about this class? it seems to be common use case to have status (similar as with `Email`), so it looks like we could have generic `Status`.

Reply:
Yes it can be reused
And now I'm cornered because I said "We'll deal with ValueObject common accross domains when we see one"
And here's one 😅 ...